numpy/core/shape_base.py

```diff
@@ -435,11 +435,6 @@ def _block(arrays, max_depth, result_ndim):
     `arrays` and the depth of the lists in `arrays` (see block docstring
     for details).
     """
-    def atleast_nd(a, ndim):
```

Comment:
It can be argued that this removal alone has lessened the overall maintenance burden on the project enough to justify this PR.